<launch>

  <node name="controller"
        pkg ="joy"
        type="joy_node"
        respawn="false"
        output="screen">
    <remap from="/joy"
             to="/hyperion/joy"/>
    <param name="dev" value="/dev/input/js0"/>
  </node>

 <node name="hyperion"
       pkg ="hyperion_straight_line"
       type="hyperion_straight_line"
       respawn="false"
       output="screen">
    <param name="kp"            type="double"      value="0.001"/>
    <param name="ki"            type="double"      value="0.00012"/>
    <param name="kd"            type="double"      value="0.00035"/>
    <param name="pid_freq"      type="double"      value="500"/>
    <param name="max_speed"     type="int"         value="100"/>
    <param name="turn_speed_saturation" type="int" value="-100"/>
    <param name="stop_button"   type="int"         value="5"/>
    <param name="pid_tunning"   type="int"         value="0"/>
    <param name="tunning_inc"   type="double"      value="0.0001"/>
    <param name="pid_mean_size" type="int"         value="19"/>
    <param name="error_calculation" type="int"     value="0"/>
    <param name="center_dist"   type="double"      value="270.0"/>
  </node>

  <node name="ultrasonic_sensors"
        pkg ="hyperion_ultrasound"
        type="ultrasonidos.py"
        respawn="false"
        output="screen">
    <remap from="/stop_topic"
           to="/hyperion/stop"/>
    <remap from="/ultrasonido_topic"
           to="/hyperion/distances"/>
  </node>

  <node name="motor_driver"
        pkg ="hyperion_motor_driver"
        type="motor_driver.py"
        respawn="false"
        output="screen">
    <remap from="/motor_speeds"
           to="/hyperion/speeds"/>
  </node>


</launch>
